File: app-mobile/src/api/links.ts
import { apiClient } from './client';

export interface PaymentLink {
  id: string;
  token: string;
  amount: string | null;
  currency: string;
  description: string;
  expires_at: string | null;
  max_uses: number | null;
  uses_count: number;
  status: string;
  checkout_url: string;
  created_at: string;
}

export interface CreateLinkParams {
  amount: number;
  description: string;
  currency?: string;
  max_uses?: number;
  expires_at?: string;   // ISO 8601
}

export async function createLink(params: CreateLinkParams): Promise<PaymentLink> {
  const { data } = await apiClient.post<PaymentLink>('/v1/links', {
    amount: params.amount,
    description: params.description,
    currency: params.currency ?? 'USD',
    max_uses: params.max_uses ?? 1,
    expires_at: params.expires_at,
  });
  return data;
}

export async function getLinkById(linkId: string): Promise<PaymentLink> {
  const { data } = await apiClient.get<PaymentLink>(`/v1/links/${linkId}`);
  return data;
}

File: app/schemas/links.py
from decimal import Decimal
from datetime import datetime
from pydantic import BaseModel, Field


class PaymentLinkCreate(BaseModel):
    amount: Decimal | None = Field(
        default=None, gt=0,
        description="Monto fijo. Si es null el cliente ingresa el monto."
    )
    currency: str = Field(default="USD", pattern="^[A-Z]{3}$")
    description: str = Field(..., min_length=1, max_length=255)
    expires_in_hours: int | None = Field(
        default=None, gt=0,
        description="Horas hasta que el link expira. Alternativa a expires_at."
    )
    expires_at: datetime | None = Field(
        default=None,
        description="Fecha/hora exacta de expiración. Alternativa a expires_in_hours."
    )
    max_uses: int | None = Field(
        default=None, gt=0,
        description="Máximo de usos. Null = ilimitado."
    )


class PaymentLinkResponse(BaseModel):
    id: str
    merchant_id: str
    token: str
    amount: Decimal | None
    currency: str
    description: str
    expires_at: datetime | None
    max_uses: int | None
    uses_count: int
    status: str
    qr_png_url: str | None
    checkout_url: str
    created_at: datetime

    model_config = {"from_attributes": True}
